### PR TITLE
Adding Past Sponsors both backend / API and frontend

### PR DESCRIPTION
**Pull request by adamrushy**

### CODE REVIEW

File: Sources/App/Features/Past Sponsors/Transformers/PastSponsorTransformer.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        }
+        
+        guard let bucketName = Environment.get("S3_BUCKET_NAME") else {
+            fatalError("Missing 'S3_BUCKET_NAME' environment variable")
+        }
+        
+        return .init(
+            id: entity.id,
+            name: entity.name,
+            image: "https://\(bucketName).s3.eu-west-2.amazonaws.com/\(entity.image)",
```
Comment (Sherlouk) [inline Sources/App/Features/Past Sponsors/Transformers/PastSponsorTransformer.swift:17]: We could probably go back through and cleanup old examples, but we don't need to do this AWS image string interpolation any more.

We just need to return `entity.image` and then within the .leaf file itself use `#awsImage(image)`.

The awsImage tag will do the interpolation for you, aiding in re-use and simplifying the transformers.

File: Sources/App/Features/Past Sponsors/Models/Migrations/PastSponsorMigration.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Fluent
+
+class PastSponsorMigration: AsyncMigration {
```
Comment (Sherlouk) [inline Sources/App/Features/Past Sponsors/Models/Migrations/PastSponsorMigration.swift:3]: Would it be possible to update the seed.sql file with some examples of this model to make it easier to onboard?

Comment (Sherlouk) [inline Sources/App/Features/Past Sponsors/Models/Migrations/PastSponsorMigration.swift:3]: What is unique about PastSponsors that we can't achieve with a query against the main Sponsors table?

The sponsors entity already includes an "event_id" column, is past sponsors not simply "select all from sponsors where event_id is not the latest event" - a query which we can quite easily achieve?

Comment (adamrushy) [reply Sources/App/Features/Past Sponsors/Models/Migrations/PastSponsorMigration.swift:3]: Yeah, so like our discussion on Slack - I would prefer to "pick and choose" what we display here

Comment (Sherlouk) [reply Sources/App/Features/Past Sponsors/Models/Migrations/PastSponsorMigration.swift:3]: I'd still prefer that be a "visible" flag on the database, though. It feels like quite a lot of effort to manage two representations of the same data just to differentiate when and where one is shown!

**Pull request closed without merging**